### PR TITLE
Remove 'common' feature test from libast

### PR DESCRIPTION
**Pull request by siteshwar**
Save some portions of the autogenerated file from iffe

### CODE REVIEW

File: src/lib/libast/include/ast_common.h [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+#ifndef Void_t
+#	if __STD_C
+#		define Void_t		void
+#	else
+#		define Void_t		char
+#	endif
+#endif
+
+/* windows variants and veneers */
+#if !defined(_WINIX) && (_UWIN || __CYGWIN__ || __EMX__)
```
Comment (catull) [inline src/lib/libast/include/ast_common.h]: This should be shortened to
```c
#if !defined(_WINIX) && (__CYGWIN__)
```
Reason: Support for UWIN and __EMX__ (.i.e OS/2) is removed now.

File: src/lib/libast/include/ast_common.h [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+#			define _DLL		1
+#		endif
+#	endif
+#	if !defined(__EXPORT__) && _BLD_DLL
+#		define __EXPORT__	__declspec(dllexport)
+#	endif
+#	if !defined(__IMPORT__) && ( _BLD_DLL || defined(_DLL) )
+#		define __IMPORT__	__declspec(dllimport)
+#	endif
+#	if _BLD_DLL && _UWIN
```
Comment (catull) [inline src/lib/libast/include/ast_common.h]: Those 3 lines are also not required anymore.
Reason: Support for UWIN is removed.

File: src/lib/libast/include/ast_common.h [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+#	endif
+#endif
+#if !defined(_astimport)
+#	if defined(__IMPORT__) && defined(_DLL)
+#		define _astimport	__IMPORT__
+#	else
+#		define _astimport	extern
+#	endif
+#endif
+#if _dll_import && ( !_BLD_DLL || _WINIX && !_UWIN )
```
Comment (catull) [inline src/lib/libast/include/ast_common.h]: Can be simplified to
```c
#if _dll_import && ( !_BLD_DLL || _WINIX)
```

**Comment by krader1961** [comment]
LGTM

**Comment by siteshwar** [comment]
@catull Thanks for the comments. I have fixed the pull request.

**Pull request merged**